orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/v1schema/graph/transform/ConfigStageInjectionTransform.java

```diff
@@ -75,17 +72,107 @@ private void replaceStages(PipelineTemplate pipelineTemplate) {
   }
 
   private void injectStages(PipelineTemplate pipelineTemplate) {
-    List<StageDefinition> configStages = templateConfiguration.getStages()
+    // Create initial graph via dependsOn.
+    createDag(pipelineTemplate.getStages());
+
+    // Handle stage injections.
+    injectStages(pipelineTemplate.getStages(), pipelineTemplate.getStages());
+    injectStages(templateConfiguration.getStages(), pipelineTemplate.getStages());
+  }
+
+  private static void createDag(List<StageDefinition> stages) {
+    stages
```

Comment:
The transforms are only intended to get the graph to the configured state, not really validate that it's acceptable. There's a validation step that we haven't added in yet that will perform validation that this would be included in: https://github.com/spinnaker/orca/blob/master/orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/PipelineTemplatePipelinePreprocessor.java#L78

Things like the `orca-validation` module (which does JSON schema validation of required fields, etc) would probably happen here, as well as doing this cyclic detection. I'm quite bad at graph stuff, so if you wanna take on the cycle detection, 💯 💯 💯 . :)